godo.go

```diff
@@ -223,7 +223,9 @@ func (c *Client) Do(req *http.Request, v interface{}) (*Response, error) {
 		return nil, err
 	}
 
-	defer resp.Body.Close()
+	defer func() {
+		err = resp.Body.Close()
```

Comment:
Do we need to check this error as well? Or does moving it into the `func(){ ... }()` deal with that?